From 32e0a9e2549c43d9abc03427ba6f3b7b8c2e1407 Mon Sep 17 00:00:00 2001
From: Dave Stevenson <dave.stevenson@raspberrypi.org>
Date: Mon, 28 Jan 2019 14:40:16 +0000
Subject: [PATCH] gpu: vc4_firmware_kms: Fix up 64 bit compile
 warnings.

Resolve two build warnings with regard using incorrectly
sized parameters in logging messages on 64 bit builds.

Signed-off-by: Dave Stevenson <dave.stevenson@raspberrypi.org>
---
 drivers/gpu/drm/vc4/vc4_firmware_kms.c | 11 ++++++-----
 1 file changed, 6 insertions(+), 5 deletions(-)

--- a/drivers/gpu/drm/vc4/vc4_firmware_kms.c
+++ b/drivers/gpu/drm/vc4/vc4_firmware_kms.c
@@ -160,14 +160,14 @@ static void vc4_primary_plane_atomic_upd
 		WARN_ON_ONCE(vc4_plane->pitch != fb->pitches[0]);
 	}
 
-	DRM_DEBUG_ATOMIC("[PLANE:%d:%s] primary update %dx%d@%d +%d,%d 0x%08x/%d\n",
+	DRM_DEBUG_ATOMIC("[PLANE:%d:%s] primary update %dx%d@%d +%d,%d 0x%pad/%d\n",
 			 plane->base.id, plane->name,
 			 state->crtc_w,
 			 state->crtc_h,
 			 bpp,
 			 state->crtc_x,
 			 state->crtc_y,
-			 bo->paddr + fb->offsets[0],
+			 &fbinfo->base,
 			 fb->pitches[0]);
 
 	ret = rpi_firmware_transaction(vc4->firmware,
@@ -197,6 +197,7 @@ static void vc4_cursor_plane_atomic_upda
 	struct drm_plane_state *state = plane->state;
 	struct drm_framebuffer *fb = state->fb;
 	struct drm_gem_cma_object *bo = drm_fb_cma_get_gem_obj(fb, 0);
+	dma_addr_t addr = bo->paddr + fb->offsets[0];
 	int ret;
 	u32 packet_state[] = {
 		state->crtc->state->active,
@@ -206,13 +207,13 @@ static void vc4_cursor_plane_atomic_upda
 	};
 	WARN_ON_ONCE(fb->pitches[0] != state->crtc_w * 4);
 
-	DRM_DEBUG_ATOMIC("[PLANE:%d:%s] update %dx%d cursor at %d,%d (0x%08x/%d)",
+	DRM_DEBUG_ATOMIC("[PLANE:%d:%s] update %dx%d cursor at %d,%d (0x%pad/%d)",
 			 plane->base.id, plane->name,
 			 state->crtc_w,
 			 state->crtc_h,
 			 state->crtc_x,
 			 state->crtc_y,
-			 bo->paddr + fb->offsets[0],
+			 &addr,
 			 fb->pitches[0]);
 
 	/* add on the top/left offsets when overscan is active */
@@ -238,7 +239,7 @@ static void vc4_cursor_plane_atomic_upda
 	    fb != old_state->fb) {
 		u32 packet_info[] = { state->crtc_w, state->crtc_h,
 				      0, /* unused */
-				      bo->paddr + fb->offsets[0],
+				      addr,
 				      0, 0, /* hotx, hoty */};
 
 		ret = rpi_firmware_property(vc4->firmware,
